fineract-provider/src/main/java/org/apache/fineract/infrastructure/jobs/service/StuckJobExecutorServiceImpl.java

```diff
@@ -36,17 +45,30 @@
 @RequiredArgsConstructor
 public class StuckJobExecutorServiceImpl implements StuckJobExecutorService {
 
+    private final FineractProperties properties;
     private final JobExecutionRepository jobExecutionRepository;
     private final TransactionTemplate transactionTemplate;
     private final JobOperator jobOperator;
+    private final JobExplorer jobExplorer;
+    private final JobRepository jobRepository;
 
     @Override
     public void resumeStuckJob(String jobName) {
-        List<Long> stuckJobIds = getStuckJobIds(jobName);
-        if (isPartitionedJob(jobName) && areThereStuckJobs(jobName)) {
-            restartPartitionedJobs(jobName, stuckJobIds);
-        } else {
-            restartTaskletJobs(stuckJobIds);
+        if (areThereStuckJobs(jobName)) {
+            final List<Long> stuckJobIds = getStuckJobIds(jobName);
+            // If It is a Batch Manager
+            if (properties.getMode().isBatchManagerEnabled()) {
```

Comment:
I dont think this is correct.  It will mark complete all the business steps that are executing on the batch workers. 

Checking whether we are the batch manager is useless. The stuck job executor only be called in batch manager.

Here what we need is to decide whether the stuck batch steps are started on the batch manager or on the batch worker. If it is a batch manager steps, and it is STARTED. We can mark them as FAILED hence the batch manager was restarted. If the found batch steps are batch worker business steps, the batch manager shall wait till they got finished before restarting the job.

Reply:
Done! It has been updated